src/modules/evas/engines/gl_generic/meson.build

```
@@ -45,10 +51,12 @@ foreach src : common_engine_src
   engine_src += files(join_paths('..','gl_common', src))
 endforeach
 
-pub_eo_files = [
-  'evas_ector_gl_buffer.eo',
-  'evas_ector_gl_image_buffer.eo'
-]
+if get_option('ector')
+  pub_eo_files = [
+    'evas_ector_gl_buffer.eo',
+    'evas_ector_gl_image_buffer.eo'
+  ]
+endif
```

Shouldn't `pub_eo_files` be declared empty for the case where there is no ector?
```suggestion
pub_eo_files = []
if get_option('ector')
  pub_eo_files += [
    'evas_ector_gl_buffer.eo',
    'evas_ector_gl_image_buffer.eo'
  ]
endif
```

Or else, the  `foreach` at line61 will yell that there are no `pub_eo_files`, right?

---

Yes, or worse: it may get a previously defined `pub_eo_files`.